### ISSUE TITLE
HTTPS Traffic

### ISSUE THREAD

**Issue by sgalvez** [buger/gor#85]
Am I able somehow to replicate https traffic?, just did a simple test and could not replicate it.

Regards.

**Comment by ramnes** [comment]
I feel like this is a real issue. Couldn't `ssldump` be used with an option?

**Comment by buger** [comment]
@ramnes ssldump is quite hacky tool, and did not updated for a while. If you really want to use it, its possible to build input plugin which will parse ssldump output, but i doubt it could work good enough.

Building input plugin should not be a problem, you can try to hack it if you want, and i provide feedback, but i do not have plans in near time for this functionality. I checked ssldump source, and it will be too much work to implement same in Gor.

**Comment by ramnes** [comment]
I've suggested `ssldump` quite randomly. Do you see anything else that could do the job in a less "hacky" way?

**Comment by buger** [comment]
Frankly not, SSL was made to protect from traffic capture, even if you have a private key, i guess thats why not so many alternatives to ssldump :)

**Comment by ramnes** [comment]
I ended up doing it with nginx by terminating SSL, doing a `proxy_pass http://localhost:8800` and then using that 8800 socket.

**Comment by buger** [comment]
:+1:

**Comment by lovewhll** [comment]
@ramnes can you show the details of HTTPS Traffic solution?

Thanks

**Comment by ramnes** [comment]
@lovewhll, there you go:

```nginx
# Force HTTPS
server {
  listen 80;
  server_name yourdomain.com;
  rewrite ^ https://$server_name$request_uri? permanent;
}

# SSL termination
server {
  listen 443 ssl;
  server_name yourdomain.com;

  access_log /var/log/nginx/access.log main;
  error_log /var/log/nginx/error.log error;

  ssl_certificate /etc/ssl/nginx/yourdomain.com.crt;
  ssl_certificate_key /etc/ssl/nginx/yourdomain.com.key;

  # ramnes: We use a socket on the port 8080 as a SSL terminated socket rather
  # than using the 443 socket directly. This is useful for us since we want to
  # to listen on that socket in clear HTTP to be able to clone packets.
  location / {
    proxy_set_header Host $host;
    proxy_pass http://localhost:8800;
  }
}

# Clear HTTP
# ramnes: here you implement your "real" nginx configuration
server {
  listen 8800;
  server_name yourdomain.com;

  root /var/www;

  location /index.html {
    if_modified_since off;
    etag off;
    expires -1;
  }

  location / {
    try_files $uri/ $uri /index.html;
  }
}
```

**Comment by lovewhll** [comment]
@ramnes Thank you!

**Comment by gmontard** [comment]
Yep this is the right way to do it, reverse proxy works pretty fine.

Anyway I just submitted an issue that could improve this setup by having Gor listen directly to a Unix Socket as an input, which would be faster than TCP Socket #102

**Comment by fifthaxe** [comment]
i suppose this topic is now to be closed since we can do this using --inputcp-secure ?

**Comment by AbdullahMuhammad20** [comment]
@buger Hi, can I ask about how I can monitor the requests after redirecting to apply on another URL when I used output-HTTP?
or how I can use the compare or Analytics? 
are we have any comments to show the differences or appear count of requests or dashboard or something like that?

**Comment by gitSambhal** [comment]
Try the following for https
```gor --input-raw :80 --input-raw-engine "raw_socket" --output-http "http://staging.com"```
